src/passes/bytesConverter/replaceIndexAccessBytesConverter.ts

```diff
@@ -0,0 +1,72 @@
+import { FixedBytesType, generalizeType, TypeName, IndexAccess, Literal } from 'solc-typed-ast';
+import { AST } from '../../ast/ast';
+import { ASTMapper } from '../../ast/mapper';
+import { createCallToFunction } from '../../utils/functionGeneration';
+import { generateExpressionTypeString } from '../../utils/getTypeString';
+import { typeNameFromTypeNode } from '../../utils/utils';
+import {
+  createNumberLiteral,
+  createUint8TypeName,
+  createUint256TypeName,
+} from '../../utils/nodeTemplates';
+import { safeGetNodeType } from '../../utils/nodeTypeProcessing';
+import { replaceBytesType } from './utils';
+
+export class ReplaceIndexAccessBytesConverter extends ASTMapper {
+  visitIndexAccess(node: IndexAccess, ast: AST): void {
+    const a = generalizeType(safeGetNodeType(node.vBaseExpression, ast.inference))[0];
+    if (!(node.vIndexExpression && a instanceof FixedBytesType)) {
+      this.visitExpression(node, ast);
+      return;
+    }
+
+    const baseTypeName = typeNameFromTypeNode(
+      safeGetNodeType(node.vBaseExpression, ast.inference),
+      ast,
+    );
+
+    const width: string = baseTypeName.typeString.slice(5);
```

Comment:
Why 5? Don't use a single number, use what your expecting to remove, like:
```ts
const width: string = baseTypeName.typeString.slice("uint".length)
```
This way is easier to understand

Reply:
Code was copy pasted from the former approach but is a valid comment, it's fixed now